engine/classes/ElggFile.php

```diff
@@ -80,6 +80,18 @@ public function getFilenameOnFilestore() {
 	}
 
 	/**
+	 * Returns path relative to the filestore (data directory) root
+	 *
```

Comment:
Instead the FileService\File class can just manually verify that filenameOnFilestore() gives back a path containing dataroot and give back a 404 if not...